src/server/search/doc_index.cc

```diff
@@ -106,8 +106,13 @@ vector<SerializedSearchDoc> ShardDocIndex::Search(const OpArgs& op_args,
     CHECK(it) << "Expected key: " << key << " to exist";
     auto doc_access = GetAccessor(op_args, (*it)->second);
     out.emplace_back(key, doc_access->Serialize());
+
+    // Scoring is not implemented yet, so we take just the first documents
+    if (out.size() >= params.limit_offset + params.limit_total)
```

Comment:
Is adding them just a temporary solution (scoring aside)?

Reply:
Yes, its temporary... But correctly computing offsets that are bound to one result set, which is computed on multiple shards, might sadly be actually impossible. Need to see later